plutarch-test/conditional/Plutarch/TryFromSpec.hs

```diff
@@ -487,7 +489,9 @@ theField = unTermCont $ do
 
 ------------------- Sample usage DerivePNewType ------------------------------------
 
-newtype PWrapInt (s :: S) = PMkWrapInt (Term s PInteger)
-  deriving (PlutusType, PIsData, PEq, POrd) via (DerivePNewtype PWrapInt PInteger)
+newtype PWrapInt (s :: S) = PWrapInt (PInteger s)
```

Comment:
different newtyping style?

Reply:
I tried to do this to keep the derive intact but it didn't work.

Reply:
Could it be changed back then? To something like this potentially:-

```hs
newtype PWrapInt (s :: S) = PWrapInt (Term s PInteger)
  deriving (PlutusType, PIsData, PEq, POrd) via (DerivePNewtype PWrapInt PInteger)

instance PTryFrom PData (PAsData PWrapInt) where
  type PTryFromExcess PData (PAsData PWrapInt) = PTryFromExcess PData (PAsData PInteger)
  ptryFrom' t f = ptryFrom' t $ \(t', exc) -> f (pouterData t', exc)

pouterData :: (PIsData a, PIsData (PInner a POpaque)) => Term s (PAsData (PInner a POpaque)) -> Term s (PAsData a)
pouterData = pdata . punsafeDowncast . pfromData
```

Though there should really be an easy deriver for cases like this. I'm fine it naming it "UnsafeNewtypeDeriver" or something but it should be there.

Reply:
Given that we will have to redo it in #441 anyway, I propose we leave it to another PR.

Reply:
wrt. the other part, why change it back?

Reply:
also pouterData there is unsafe

Reply:
If a new style is unnecessary, it should just use the same newtyping style as other places in the codebase - for the sake of consistency.

`pouterData` is indeed unsafe in the general sense, which is why I suggested "UnsafeNewtypeDeriver". It's morally safe in this invariant-less case though.

Reply:
I think it's better to do it this way since we don't need punsafeCoerce that way.

Reply:
actually wait...

Reply:
currently `type role PAsData representational phantom`, so if `Coercible a b`, you have `Coercible (PAsData a) (PAsData b)`, which is in fact completely incorrect! lol

Reply:
I will fix this and then I think we can merge so we can work on #446 ?

Reply:
Time for a release cycle sprint within a release cycle sprint. I hope we come out the other side with across-the-board improvements.